### PR TITLE
Dynamic detection of CRDs

### PR DESCRIPTION
**Pull request by jojimt**
- Added infra for dynamically detecting CRDs. This means the corresponding handler code is only invoked if the CRD is detected
 - Moved qos handling code to this new mode as reference

### CODE REVIEW

File: pkg/controller/crds.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func (cont *AciController) initCRDInformerBase(listWatch *cache.ListWatch) {
+	_, cont.crdInformer = cache.NewInformer(
+		listWatch, &v1.CustomResourceDefinition{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				crd, ok := obj.(*v1.CustomResourceDefinition)
+				if !ok {
+					cont.log.Errorf("Add CRD bad object")
```
Comment (jayaramsatya) [inline pkg/controller/crds.go:48]: can we return here ? as we are accessing  the object in crdAdded

File: pkg/controller/qos.go [hunk 2]
```diff
@@ -47,6 +51,20 @@ func QosPolicyLogger(log *logrus.Logger, qos *qospolicy.QosPolicy) *logrus.Entry
...
+func qosInit(cont *AciController, stopCh <-chan struct{}) {
+	cont.log.Debug("Initializing qos client")
+	restconfig := cont.env.RESTConfig()
+	qosClient, err := qosclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize qos client")
+		return
+	}
+	cont.initQosInformerFromClient(qosClient)
+	cache.WaitForCacheSync(stopCh, cont.qosInformer.HasSynced)
```
Comment (jayaramsatya) [inline pkg/controller/qos.go:63]: can we move the informer run before the waitForCache Sync

Comment (jojimt) [reply pkg/controller/qos.go:63]: I think I will eliminate the WaitForCacheSync. It doesn't serve any purpose in this workflow actually.

File: pkg/controller/crds.go (hunk 1 above)
Comment (jojimt) [reply pkg/controller/crds.go:48]: Yes, good catch.

**Review by jayaramsatya — APPROVED** [review]
LGTM

**Pull request merged**